### ISSUE TITLE
Mobile Support Broken

### ISSUE THREAD

**Issue by wlpdrpat** [Automattic/wp-super-cache#152]
This is tested on clean install of wp multisite 4.6.1 and latest version of wp super cache. No other plugins active and using default theme. Issue persists for both mod_rewrite and php caching configured.

Index-mobile.html (gz) is only generated if mobile device is used and index.html (gz) does not exist. If index.html(gz) exists it is delivered for both mobile and desktop browsers and index-mobile.html (gz) is not generated.

**Comment by kraftbj** [comment]
Thanks for the report. We'll check it out.

**Comment by wlpdrpat** [comment]
@kraftbj My team worked through the night debugging this issue. We found that when serving the cached version it is not checking for mobile device via wp_cache_check_mobile function and that appears to be the cause of the issue. We are continuing to review the code and will let you know if we find a solution.

**Comment by Optimized-Marketing** [comment]
We are having the same issue with Mobile Support Not Loading the Mobile Cache. It worked fine until WP 4.6.1. This issue might be a similar to https://github.com/Automattic/wp-super-cache/issues/132

**Comment by wlpdrpat** [comment]
@Optimized-Marketing Thank you very much for the suggestion. I saw that thread but didn't read through it as I thought it related more to Jetpack than WP Super Cache. I will test and let you know if it works in our case.

**Comment by Optimized-Marketing** [comment]
The solution in there doesn't full solve this problem, it helps to loads the mobile pages but it recreates the mobile cache file every time.

**Comment by wlpdrpat** [comment]
@Optimized-Marketing thank you for the details. I just finished testing it and found exactly what you have described. As such I have reverted to my temporary fix of excluding mobile user agents as it uses less server resources. I simply added a list of mobile user agents to the Exclude User Agents on the Advanced tab. By doing this it serves the mobile theme without using cache. Definitely not an ideal solution but better than continually recreating the mobile cache.

**Comment by wlpdrpat** [comment]
@kraftbj @Optimized-Marketing I found a fix (at least for my situation). I implemented the changes suggested in #132 and as I previously mentioned it didn't fix the problem. However after making that change I then used the Restore Default Settings on the Advanced tab and then re-applied the settings we normally use. Eureka! It appears to be working as expected. This worked for both my testing installation and my production install.

I think something must have been modified in the wp-cache-config.php that didn't get updated when I upgraded. Using the Restore Default Settings must have fixed/modified something about that file that allowed the upgraded version to work.

A couple related issue that I still noted: 

1. If Don't Cache pages for known users is not checked then the problem persists for known users (I think this is because it serves php cache to known users).
2. mod_rewrite must be selected as php cache does not work as expected - it gives a similar problems as previously described.

Thanks,
Pat

**Comment by jchuckmailen** [comment]
@kraftbj @Optimized-Marketing @wlpdrpat This is the temporary fix we're currently using for PHP caching, which properly serves the mobile file and does not regenerate each time the page loads.

We've added a file in the WP Super Cache PlugIns folder with the following code:

`
add_filter( 'supercache_filename_str', 'correct_mobile_cache_issue', 100, 1 );
function correct_mobile_cache_issue( $str ) {
	$is_mobile = wp_cache_check_mobile();
	$str .= $is_mobile;
	$str = str_replace( '-mobile-mobile', '-mobile', $str );
	return $str;
}
`

It's an ugly fix, but it's functional.

From our testing it appears that the mobile cache page is being built, but there seems to be a mobile check missing when filtering the filename in phase1, so instead of serving the mobile file it serves the desktop file.

The above fix runs a mobile check and adds '-mobile' if a mobile device is detected. However we discovered that other parts of the code DO properly add the '-mobile' to the filename, so we added that str_replace to make sure '-mobile' isn't added to the filename twice.

As mentioned, this is a dirty fix, but it might help someone else with a temporary solution.

If there's time this month we might dig into WP Super Cache a little deeper and see if we can come up with a viable pull request to help resolve this issue.
